src/Mod/Fem/femsolver/elmer/equations/electrostatic_writer.py

```diff
@@ -125,22 +125,30 @@ def handleElectrostaticBndConditions(self):
                     # output the FreeCAD label as comment
                     if obj.Label:
                         self.write.boundary(name, "! FreeCAD Name", obj.Label)
-                    if obj.PotentialEnabled:
-                        if hasattr(obj, "Potential"):
-                            # Potential was once a float and scaled not fitting SI units
-                            if isinstance(obj.Potential, float):
-                                savePotential = obj.Potential
-                                obj.removeProperty("Potential")
-                                obj.addProperty(
-                                    "App::PropertyElectricPotential",
-                                    "Potential",
-                                    "Parameter",
-                                    "Electric Potential",
-                                )
-                                # scale to match SI units
-                                obj.Potential = savePotential * 1e6
-                            potential = float(obj.Potential.getValueAs("V"))
-                            self.write.boundary(name, "Potential", potential)
+                    if not hasattr(obj, "Diriclet") or obj.Diriclet == True:
+                        if obj.PotentialEnabled:
+                            if hasattr(obj, "Potential"):
+                                # Potential was once a float and scaled not fitting SI units
+                                if isinstance(obj.Potential, float):
+                                    savePotential = obj.Potential
+                                    obj.removeProperty("Potential")
+                                    obj.addProperty(
+                                        "App::PropertyElectricPotential",
+                                        "Potential",
+                                        "Parameter",
+                                        "Electric Potential",
+                                    )
+                                    # scale to match SI units
+                                    obj.Potential = savePotential * 1e6
+                                potential = float(obj.Potential.getValueAs("V"))
+                                self.write.boundary(name, "Potential", potential)
+                    elif (
+                        hasattr(obj, "Diriclet")
+                        and obj.Diriclet == False
+                        and hasattr(obj, "ElectricFlux")
+                    ):
+                        flux = float(obj.ElectricFlux.getValueAs("A*s/m^2"))
+                        self.write.boundary(name, "Electric Flux", flux)
```

Comment:
I was not aware of that mechanism, and will try to tidy this up.

I think the changes to potential originated from someone else than me. It might be that it was fixed upstream in a later commit, and then reappeared here due to merging. I'll check.

Reply:
I'm reworking the electrostatic constraint file and it has some impact on this file as well. If you want, leave this as is. I'll rebase your changes and then fix this so you don't have to do the work again.